### DISCUSSION TITLE
Does Thonny migrate pip packages while upgrading?

### DISCUSSION THREAD

**Discussion by Welding-Torch** [thonny/thonny#3166] [General]
This is a question for the maintainers. I updated Thonny a while back. And I think in that update, it shifted from Python 3.7 to Python 3.10. That's good. But today I found a folder called Python37 in my AppData folder. And it had all the old pip packages I installed there.

So, does Thonny migrate pip packages while upgrading? I think it doesn't. Can it be fixed? 

Can you make it so that pip packages from the old version are exported into a list, then that list is fed to the new pip, and then the packages are all installed again, and the old python (In my case Python37) folder is deleted?

This saves space.

Correct me if I have made any mistakes.

Thanks.

**Comment by aivarannamaa** [comment]
If you are not using virtual environments, then Thonny installs packages with `pip install --user <package name>`. This means, the packages get installed per-user but system-wide. If Thonny's Python is version 3.7, then the newly installed package becomes available for every Python 3.7 installation in your system. Removing these packages afterwards may break something else.

I've considered different options, but they get weird in other situations. Python package management is cumbersome.

**Comment by Welding-Torch** [comment]
> Python package management is cumbersome.  

Ain't that the truth

But my question still isn't answered actually. If I upgrade to a newer version of thonny(which also means a newer version of python), then will the packages from my old thonny installation be transferred to the new one?

If there's no system in place to do this already, can you implement it?

**Reply by aivarannamaa** [reply]
No, Thonny does not transfer installed packages and I don't want to implement it because of the aforementioned reasons. In short -- user site packages can affect several Python installations and any automatic changes could cause confusion for some users.

**Reply by Welding-Torch** [reply]
However, it causes confusion for users updating Thonny because it breaks their existing code which relies on packages they installed.
